report_alternative_layout/models/report_paperformat.py

```diff
@@ -8,11 +8,13 @@ class ReportPaperformat(models.Model):
     _inherit = "report.paperformat"
 
     apply_alternative_layout = fields.Boolean(
+        company_dependent=True,
```

Comment:
IMO, it's a design flaw to make fields in the paperformat model company-dependent. Instead, we should make the paperformat_id of the report company-dependent. It should ideally be done in the Odoo core but it will only be accepted in master. Then a sane approach for stable versions may be to create a new module in the reporting-engine repo.

Reply:
I tried to check the behavior of using company_dependent for paperformat_id, but I realized that the existing value is only applied to the main company. For other companies, the value will be None.
To handle this, we could use a post-init hook to assign the value. However, if a new company is created after the module is installed, we would need to assign the value for all reports again.
I think this is not ideal, even if it’s a rare case. Can we keep the current design, considering that it's uncommon to use different paper formats across multiple companies, and the differences are usually minor, such as showing or hiding the remit bank information?